malloc/malloc.c

```diff
@@ -26,7 +35,14 @@ char *string_dup(char *src)
 */
 void *mem_copy(void *dest, const void *src, int n)
 {
+ const char *csrc = (char *) src;
+char *cdest = (char *) dest;
+
+for (int i = 0; i < n; i++) {
+    *(cdest + i) = *(csrc + 1);
+}
 
+return dest;
 }
 
 /*
```

Comment:
This partially runs on my system but eventually seg faults. It's a good start, though!

Reply:
This no longer seg faults but it still does not output correctly. I am getting a "control reaches end of non-void function" error. This might mean that you're getting too close to the end in at least one of your loops and maybe should be using i<n-1 instead of i<n or something.